crates/lang/codegen/src/generator/dispatch.rs

```diff
@@ -114,6 +115,40 @@ impl Dispatch<'_> {
             .count()
     }
 
+    /// Returns the index of the ink! message which has a wildcard selector, if existent.
+    fn query_wildcard_message(&self) -> Option<usize> {
+        let mut wildcard_selectors = self
+            .contract
+            .module()
+            .impls()
+            .map(|item_impl| item_impl.iter_messages())
+            .flatten()
+            .positions(|item| item.has_wildcard_selector());
+        let next = wildcard_selectors.next();
+        assert!(
+            wildcard_selectors.next().is_none(),
+            "found more than one wildcard selector message, contracts must have at most one."
+        );
+        next
+    }
+
+    /// Returns the index of the ink! constructor which has a wildcard selector, if existent.
+    fn query_wildcard_constructor(&self) -> Option<usize> {
+        let mut wildcard_selectors = self
+            .contract
+            .module()
+            .impls()
+            .map(|item_impl| item_impl.iter_constructors())
+            .flatten()
+            .positions(|item| item.has_wildcard_selector());
+        let next = wildcard_selectors.next();
+        assert!(
+            wildcard_selectors.next().is_none(),
+            "found more than one wildcard selector constructor, contracts must have at most one."
+        );
+        next
```

Comment:
```suggestion
            .position(|item| item.has_wildcard_selector())
```

Reply:
We don't need to assert anything in the `ink_codegen` module since it relies on the fact that the whole semantics checking are already done by the `ink_ir` layer. This way `ink_codegen` transformation are kind of infallible ideally.